### PR TITLE
Add model classes for building board

### PR DESCRIPTION
**Pull request by patiem**

### CODE REVIEW

File: client/src/main/java/model/BoundField.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import javafx.scene.paint.Color;
+import javafx.scene.shape.Rectangle;
+import model.Field;
+
+
+public class BoundField extends Rectangle implements Field {
+
+    public BoundField(int i, int n) {
+        super(i, n, 30, 30);
```
Comment (robwrobel) [inline client/src/main/java/model/BoundField.java]: magic numbers

File: client/src/main/java/model/BoundariesPosition.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    public BoundariesPosition(Mast mast) {
+        this.mast = mast;
+    }
+
+    public List<Position> countBoundariesForMast() {
+
+        List<Position> boundaries = new ArrayList<>();
+
+        for (int x = -1; x < 2; x++) {
```
Comment (robwrobel) [inline client/src/main/java/model/BoundariesPosition.java]: try to make it simpler

File: client/src/main/java/model/FieldPosition.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package model;
+
+public class FieldPosition {
+
+
+    private final Integer x;
+    private final Integer y;
+
+    public FieldPosition(double x, double y) {
+        this.x = (int) x;
```
Comment (robwrobel) [inline client/src/main/java/model/FieldPosition.java]: refactor

File: client/src/main/java/model/Fleet.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Fleet {
+
+
+    private List<Mast> mastList;
+    private Sea sea;
+    private List<Ship> flota;
```
Comment (robwrobel) [inline client/src/main/java/model/Fleet.java]: fleet

File: client/src/main/java/model/Fleet.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
+package model;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Fleet {
+
+
+    private List<Mast> mastList;
```
Comment (robwrobel) [inline client/src/main/java/model/Fleet.java]: masts

File: client/src/main/java/model/Fleet.java [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    private Sea sea;
```
Comment (robwrobel) [inline client/src/main/java/model/Fleet.java]: final

File: client/src/main/java/model/Fleet.java [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+    public void buildShip(Mast mast) {
+        add(mast);
+        Ship ship = new Ship(mast);
+        lastShip = ship;
+        flota.add(ship);
+        createNewMast(mast);
+    }
+
+    private void createNewMast(Mast mast) { //TODO: this methods has two responsibilities
```
Comment (robwrobel) [inline client/src/main/java/model/Fleet.java]: remove TODO

File: client/src/main/java/model/Fleet.java [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+
+    public Fleet(Sea sea) {
+        this.sea = sea;
+        mastList = new ArrayList<>();
+        flota = new ArrayList<>();
+    }
+
+    void add(Mast mast) {
+        System.out.println(mast);
```
Comment (robwrobel) [inline client/src/main/java/model/Fleet.java]: consider adding logger

File: client/src/main/java/model/Fleet.java [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+        if(lastShip.isShipDone()) {
+            sea.clearSea();
+            sea.makeBoundaries(lastShip);
+            return;
+        }
+        PossiblePosition possible = new PossiblePosition(mast, sea);
+        possible.showPossibleMastPosition();
+    }
+
+    public void glueMast(Mast mast) {
```
Comment (robwrobel) [inline client/src/main/java/model/Fleet.java]: rename

File: client/src/main/java/model/Mast.java [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+import javafx.scene.paint.Color;
+import javafx.scene.shape.Rectangle;
+import model.Field;
+
+public class Mast extends Rectangle implements Field {
+
+    private int length;
+
+    public Mast(int i, int n, int lenght) {
+        super(i, n, 30, 30);
```
Comment (robwrobel) [inline client/src/main/java/model/Mast.java]: magic numbers

File: client/src/main/java/model/Mast.java [hunk 11]
```diff
@@ -0,0 +1,21 @@
...
+
+import javafx.scene.paint.Color;
+import javafx.scene.shape.Rectangle;
+import model.Field;
+
+public class Mast extends Rectangle implements Field {
+
+    private int length;
+
+    public Mast(int i, int n, int lenght) {
```
Comment (robwrobel) [inline client/src/main/java/model/Mast.java]: i and n rename

File: client/src/main/java/model/Position.java [hunk 12]
```diff
@@ -0,0 +1,60 @@
+package model;
+
+import java.util.Objects;
+
+public class Position {
+
+    Integer x;
```
Comment (robwrobel) [inline client/src/main/java/model/Position.java]: private final

File: client/src/main/java/model/Position.java [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+    Integer y;
+
+    public Position(int x, int y) {
```
Comment (robwrobel) [inline client/src/main/java/model/Position.java]: add meaningfull names

File: client/src/main/java/model/PossiblePosition.java [hunk 14]
```diff
@@ -0,0 +1,23 @@
...
+    public PossiblePosition(Mast mast, Sea sea) {
+        this.mast = mast;
+        this.sea = sea;
+    }
+
+    void showPossibleMastPosition() {
+        System.out.println(mast.getX());
+        System.out.println(mast.getY());
+        if(mast.getY() >= 1 ) sea.getSeaFieldByPosition(Position.up(mast.position())).makeToClick();
+        if(mast.getY() <= 8 ) sea.getSeaFieldByPosition(Position.down(mast.position())).makeToClick();
```
Comment (robwrobel) [inline client/src/main/java/model/PossiblePosition.java]: magic number

File: client/src/main/java/model/PossiblePosition.java [hunk 15]
```diff
@@ -0,0 +1,23 @@
+package model;
+
+public class PossiblePosition {
+
+    Mast mast;
```
Comment (robwrobel) [inline client/src/main/java/model/PossiblePosition.java]: private final ?

File: client/src/main/java/model/SeaField.java [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+import javafx.scene.paint.Color;
+import javafx.scene.shape.Rectangle;
+
+public class SeaField extends Rectangle implements Field {
+
+    SimpleBooleanProperty isMarkedAsMast;
+    SimpleBooleanProperty isMarkedAsBound;
+
+    public SeaField(int i, int n) {
+        super(i, n, 30, 30);
```
Comment (robwrobel) [inline client/src/main/java/model/SeaField.java]: magic

File: client/src/main/java/model/SeaField.java [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+        setFill(Color.AZURE);
+        setStroke(Color.GRAY);
+        isMarkedAsMast = new SimpleBooleanProperty(this, "mast", false);
+        isMarkedAsBound = new SimpleBooleanProperty(this, "bound", false);
+    }
+
+    public void hit() {
+        setFill(Color.RED);
+        System.out.println(getX());
+        System.out.println(getY());
```
Comment (robwrobel) [inline client/src/main/java/model/SeaField.java]: remove

File: client/src/main/java/model/SeaField.java [hunk 18]
```diff
@@ -0,0 +1,81 @@
...
+        setFill(Color.BLACK);
+    }
+
+    public void makeToClick() {
+        setFill(Color.BLANCHEDALMOND);
+        this.addEventHandler(MouseEvent.MOUSE_CLICKED, makeReadyToClick);
+    }
+
+    public Integer calculateListPosition() {
+        return (int) (getX() + getY() * 10);
```
Comment (robwrobel) [inline client/src/main/java/model/SeaField.java]: magic

File: client/src/main/java/model/SeaField.java [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+        return isMarkedAsBound;
+    }
+
+    public void setIsMarkedAsBound(boolean isMarkedAsBound) {
+        this.isMarkedAsBound.set(isMarkedAsBound);
+    }
+
+    EventHandler<MouseEvent> makeReadyToClick = event -> {
+                SeaField field = (SeaField) event.getSource();
+                System.out.println(field.getX());
```
Comment (robwrobel) [inline client/src/main/java/model/SeaField.java]: remove

File: client/src/main/java/model/Ship.java [hunk 20]
```diff
@@ -0,0 +1,43 @@
...
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class Ship {
+
+    private Mast[] masts;
+    private int length;
```
Comment (robwrobel) [inline client/src/main/java/model/Ship.java]: final

File: client/src/main/java/model/Ship.java [hunk 21]
```diff
@@ -0,0 +1,43 @@
...
+
+    boolean isShipDone() {
+        return buildLength == length;
+    }
+
+    void addMast(Mast mast) {
+        if (!isShipDone()) masts[buildLength++] = mast;
+    }
+
+    public List<Position> getMastsPositions() {
```
Comment (robwrobel) [inline client/src/main/java/model/Ship.java]: rename

File: client/src/main/java/model/BoundField.java (hunk 1 above)
Comment (patiem) [reply client/src/main/java/model/BoundField.java]: done

File: client/src/main/java/model/BoundariesPosition.java (hunk 2 above)
Comment (patiem) [reply client/src/main/java/model/BoundariesPosition.java]: almost done

File: client/src/main/java/model/FieldPosition.java (hunk 3 above)
Comment (patiem) [reply client/src/main/java/model/FieldPosition.java]: done

File: client/src/main/java/model/Fleet.java (hunk 5 above)
Comment (patiem) [reply client/src/main/java/model/Fleet.java]: done

File: client/src/main/java/model/Fleet.java (hunk 6 above)
Comment (patiem) [reply client/src/main/java/model/Fleet.java]: done

File: client/src/main/java/model/Fleet.java (hunk 4 above)
Comment (patiem) [reply client/src/main/java/model/Fleet.java]: done

File: client/src/main/java/model/Fleet.java (hunk 8 above)
Comment (patiem) [reply client/src/main/java/model/Fleet.java]: done

File: client/src/main/java/model/Fleet.java (hunk 9 above)
Comment (patiem) [reply client/src/main/java/model/Fleet.java]: done

File: client/src/main/java/model/Mast.java (hunk 11 above)
Comment (patiem) [reply client/src/main/java/model/Mast.java]: done

File: client/src/main/java/model/Mast.java (hunk 10 above)
Comment (patiem) [reply client/src/main/java/model/Mast.java]: done

File: client/src/main/java/model/Position.java (hunk 12 above)
Comment (patiem) [reply client/src/main/java/model/Position.java]: done

File: client/src/main/java/model/Position.java (hunk 13 above)
Comment (patiem) [reply client/src/main/java/model/Position.java]: done

File: client/src/main/java/model/PossiblePosition.java (hunk 15 above)
Comment (patiem) [reply client/src/main/java/model/PossiblePosition.java]: done

File: client/src/main/java/model/PossiblePosition.java (hunk 14 above)
Comment (patiem) [reply client/src/main/java/model/PossiblePosition.java]: done

File: client/src/main/java/model/SeaField.java (hunk 16 above)
Comment (patiem) [reply client/src/main/java/model/SeaField.java]: done

File: client/src/main/java/model/SeaField.java (hunk 18 above)
Comment (patiem) [reply client/src/main/java/model/SeaField.java]: done

File: client/src/main/java/model/Ship.java (hunk 20 above)
Comment (patiem) [reply client/src/main/java/model/Ship.java]: done

File: client/src/main/java/model/SeaField.java (hunk 17 above)
Comment (patiem) [reply client/src/main/java/model/SeaField.java]: done

File: client/src/main/java/model/BoundariesPosition.java [hunk 22]
```diff
@@ -0,0 +1,34 @@
...
+
+public class BoundariesPosition {
+
+    private Mast mast;
+
+    public BoundariesPosition(Mast mast) {
+        this.mast = mast;
+    }
+
+    public List<Position> countBoundariesForMast() {
```
Comment (EmiliaCiastek) [inline client/src/main/java/model/BoundariesPosition.java]: have to add tests

**Pull request merged**